gufe/components/explicitmoleculecomponent.py

```diff
@@ -186,3 +125,62 @@ def _from_dict(cls, d: dict):
 
     def _to_dict(self) -> dict:
         raise NotImplementedError()
+
+    def _check_partial_charges(self) -> None:
+        """
+        Checks for the presence of partial charges.
+
+        Note
+        ----
+        We ensure the charges are set as atom properties
+        to ensure they are detected by OpenFF
+
+        Raises
+        ------
+        ValueError
+        * If the partial charges are not of length atoms.
+        * If the sum of partial charges is not equal to the
+            formal charge.
+        UserWarning
+        * If partial charges are found.
+        * If the partial charges are near 0 for all atoms.
+        """
+        mol = self._rdkit
+
+        if "atom.dprop.PartialCharge" not in mol.GetPropNames():
+            return
+
+        p_chgs = np.array(mol.GetProp("atom.dprop.PartialCharge").split(), dtype=float)
+
+        if len(p_chgs) != mol.GetNumAtoms():
+            errmsg = (
+                f"Incorrect number of partial charges: {len(p_chgs)} " f" were provided for {mol.GetNumAtoms()} atoms"
+            )
+            raise ValueError(errmsg)
+
+        if abs(sum(p_chgs) - Chem.GetFormalCharge(mol)) > 0.01:
+            errmsg = (
+                f"Sum of partial charges {sum(p_chgs)} differs from " f"RDKit formal charge {Chem.GetFormalCharge(mol)}"
+            )
+            raise ValueError(errmsg)
+
+        # set the charges on the atoms if not already set
+        for i, charge in enumerate(p_chgs):
+            atom = mol.GetAtomWithIdx(i)
+            if not atom.HasProp("PartialCharge"):
+                atom.SetDoubleProp("PartialCharge", charge)
+            else:
+                atom_charge = atom.GetDoubleProp("PartialCharge")
+                if not np.isclose(atom_charge, charge):
+                    errmsg = (
+                        f"non-equivalent partial charges between atom and "
+                        f"molecule properties: {atom_charge} {charge}"
+                    )
+                    raise ValueError(errmsg)
+
+        if np.all(np.isclose(p_chgs, 0.0)):
+            wmsg = "Partial charges provided all equal to zero. These may be ignored by some Protocols."
+            warnings.warn(wmsg)
+        else:
+            msg = f"Partial charges are present for {self.key} (name: '{self.name}')"
```

Comment:
A note for others, we went with gufekey and the name of the component as writing out the smiles to the log might not be a good idea as this can be sensitive information that should not be shared.